Automattic/pocket-casts-android
# Display artwork from podcast feed via show notes endpoint on full-screen player and episodes screens

## Description
This adds support for displaying the episode-specific artwork that is sometimes included in a podcast's RSS feed. This information is available to the app from the call to retrieve the show notes that was [implemented already](https://github.com/Automattic/pocket-casts-android/pull/1083).

Fixes #528

Look at why the podcast artwork briefly displays before the episode artwork on the episode screen

## Testing Instructions

### 1. Displays episode artwork
1. Fresh install the app
3. Open up the podcast screen for "Darknet Diaries"
4. Open the episode details page for an episode
5. ✅  Observe that episode-specific  artwork is displayed instead of the podcast artwork (the first time you do this the podcast artwork may show for a split-second before the episode artwork url is available—that appears to only happen the one time though, visiting other episodes later will not show that issue).
6. Begin playing that episode
7. Go to the full-screen player
8. ✅ Observe that the episode-specific artwork is displayed in the full-screen player
9. Stop playback
10. Turn on airplane mode
11. Open the episode screen for a different episode of Darknet Diaries
12. ✅ Observe that the podcast's default artwork is displayed (we have the URL for all of the podcast's episode artwork on the device at this point, but we can't retrieve the image from the url because of airplane mode)
13. Attempt to play this episode (it will fail because of airplane mode)
14. Open the full-screen player
15. ✅  Observe that the podcast artwork is displayed
16. Go back and open the episode screen from the episode where you saw the episode artwork earlier
17. ✅ Observe that you can see the (cached) episode artwork
18. Attempt to play this episode (it will fail because of airplane mode)
19. Open the full-screen payer
20. ✅ Observe that you can see the (cached) episode artwork
21. Turn off airplane mode
22. Without going to the episode screen for the episode (i.e., from the podcast screen), play an episode you haven't played yet.
23. Without going to the episode screen for the episode, open the full-screen player
24. ✅ Observe that you can see the episode artwork
25. Attach the network inspector in Android Studio
26. Subscribe to Darknet diaries
27. Confirm that the app uses the cache and the `episodes_full_#####.json` call gets a 304 response with no body.

### 2. Artwork for user files still works
1. Begin playing a user file with an associated image
5. Verify that the file image is shown in the full-screen player

### 3. Show notes are still fetched with downloads
1. Fresh install the app
6. Open the podcast page for Darknet Diaries
7. Turn on airplane mode
8. Open an episode and observe you do not see the show notes
9. Back out to the podcast screen
10. Turn off airplane mode
11. Within the app's General Settings change the "Row action" to "Download".
12. Without visiting the episode details screen or playing the episode, but from the podcast screen, download an episode.
13. Once the download is finished, turn on airplane mode again
14. Open the episode screen for the downloaded episode
15. ✅ Observe that the show notes are available (i.e., they were fetched when we downloaded the episode).

### 4. Test the migration
1. Install the app from `main`
2. Use the app a bit, subscribe to a couple of podcasts, play a couple of episodes, etc.
3. Install the app from this PR
4. Play an episode of "Darknet diaries"
5. Confirm that the episode artwork is displayed.

## Screenshots or Screencast 

| Episode details screen | Full-screen player |
| --- | --- |
| ![Screenshot 2023-12-08 at 11 02 41 AM](https://github.com/Automattic/pocket-casts-android/assets/4656348/186e11ce-2a9e-4037-a016-06448379dbb5) | ![Screenshot 2023-12-08 at 11 02 31 AM](https://github.com/Automattic/pocket-casts-android/assets/4656348/c811e04f-067d-45ba-8f0c-fe909d7ea6c1) |

## Checklist
 
#### I have tested any UI changes...
